playbooks/fundamentals/template_development_guide.adoc

```diff
@@ -0,0 +1,387 @@
+rc/openjdk-app-1   1         1         1         21d
+rc/spring-rest-1   1         1         1         20d
+
+NAME                 HOST/PORT                                         PATH      SERVICES      PORT      TERMINATION   WILDCARD
+routes/openjdk-app   openjdk-app-eric-test.apps.d1.casl.rht-labs.com             openjdk-app   <all>                   None
+routes/spring-rest   spring-rest-eric-test.apps.d1.casl.rht-labs.com             spring-rest   <all>                   None
+
+NAME              CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+svc/openjdk-app   172.30.125.201   <none>        8080/TCP   21d
+svc/spring-rest   172.30.61.234    <none>        8080/TCP   20d
+
+NAME                     READY     STATUS      RESTARTS   AGE
+po/openjdk-app-1-build   0/1       Completed   0          21d
+po/openjdk-app-1-gwtj9   1/1       Running     0          21d
+po/spring-rest-1-build   0/1       Completed   0          20d
+po/spring-rest-1-xtbx2   1/1       Running     0          20d
+```
+
+Let's go ahead and clean up the old `openjdk-app` resources. Because the template we used to create the objects made good use of labels in its `objects` list, we can do this very easily.
+
+```
+$ oc delete all -l application=openjdk-app
+buildconfig "openjdk-app" deleted
+imagestream "openjdk-app" deleted
+deploymentconfig "openjdk-app" deleted
+route "openjdk-app" deleted
+service "openjdk-app" deleted
+pod "openjdk-app-1-gwtj9" deleted
+```
+
+So far, we've learned that...
+
+- a _Template_ is a collection of resource definitions that can be parameterized
+- `oc new-app` is a very simple and easy way to quickly instantiate a template
+- templates can be loaded into OpenShift and then referenced by name
+
+This is a great start, but it does leave some further questions that might be worth exploring:
+
+- How else could I work with templates?
+- What about templates that aren't pre-loaded into OpenShift?
+- How might I update resources that were created from a template?
+
+Let's move on to the next phase in our exploration
+
+=== Template files, processing, applying
+
+=== What we've learned and where to go from here
+
+== Building Custom Templates
+
+Custom templates allow a user to truly unlock the power of OpenShift in many ways. This section will dive into various approaches to building custom templates. But first, let's dive into the basic structure and makeup of a template.
+
+=== Template Structure
+
+The basic top level structure of an OpenShift template is as follows:
+
+```
+apiVersion: v1
+kind: Template
+labels:
+message: <Creation message>
+metadata:
+  name: <template name>
+objects:
+parameters:
+```
+
+The important sections here are:
+
+- `kind: Template` - defines the object as a template
+- `message` - An optional message to return to the user when the template is created using the Web Console
+- `metadata` - Standard metadata section for all Kubernetes objects, including object `name`.
```

Comment:
Do we want to call out the specific annotations that are that drive the UI such as tags, icon-class, display name etc?

Reply:
@sabre1041 would you want to see that as a nested list? any you can think of other than the three you mentioned?